test/ssl_test_ctx.c

```diff
@@ -480,6 +480,39 @@ __owur static int parse_expected_client_cert_type(SSL_TEST_CTX *test_ctx,
                                    value);
 }
 
+/*************************/
+/* Expected signing hash */
+/*************************/
+
+__owur static int parse_expected_sign_hash(int *ptype, const char *value)
+{
+    int nid;
+
+    if (value == NULL)
+        return 0;
+    nid = OBJ_sn2nid(value);
+    if (nid == NID_undef)
+        nid = OBJ_ln2nid(value);
+    if (nid == NID_undef)
```

Comment:
I think it's && for that specific example. We are comparing the return value to NID_undef so if lookup fails then the subexpression is true. So iff OBJ_sn2nid() returns NID_undef the expression is true and we proceed to the OBJ_ln2nid() part. If that also returnd NID_undef the whole thing is true and we return an error due to failed lookup.